Add support for text indicator style for mac.

megangardner opened the pull request:
#### bd534ffe2e5dee861e21b98296c1cf9295a1f19a
<pre>
Add support for text indicator style for mac.
<a href="https://bugs.webkit.org/show_bug.cgi?id=272121">https://bugs.webkit.org/show_bug.cgi?id=272121</a>
<a href="https://rdar.apple.com/123134304">rdar://123134304</a>

Reviewed by Wenson Hsieh.

Add support for text indicator style for mac.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _removeTextIndicatorStyleForID:]):
* Source/WebKit/UIProcess/mac/WKTextIndicatorStyleManager.h: Added.
* Source/WebKit/UIProcess/mac/WKTextIndicatorStyleManager.mm: Added.
* Source/WebKit/UIProcess/mac/WebViewImpl.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::addTextIndicatorStyleForID):
(WebKit::WebViewImpl::removeTextIndicatorStyleForID):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/277080@main">https://commits.webkit.org/277080@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cbcf6619096ccd06f45b00f3f2b1828e01dc9a32

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/46615 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/25772 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/49222 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/49292 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/42660 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/30132 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/23234 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/38012 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/47194 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/22760 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/40172 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/19287 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/20147 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/41289 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/4660 "Built successfully") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/39862 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/42922 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/41667 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/51143 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/46097 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/21622 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/18020 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/45270 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/22912 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/44221 "Passed tests") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/23360 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/53240 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6518 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/22615 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/10918 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebKit/UIProcess/mac/WKTextIndicatorStyleManager.h:
```diff
@@ -0,0 +1,50 @@
+
+
+
+
```
These lines can go

whsieh replied:
Nit - lots of extra newlines here..

whsieh commented on Source/WebKit/UIProcess/mac/WebViewImpl.mm:
```diff
@@ -4498,6 +4499,30 @@ static NSPasteboardName pasteboardNameForAccessCategory(WebCore::DOMPasteAccessC
+
+    if (!m_textIndicatorStyleManager)
+        return;
+
```
I don't think you need this early return.
```suggestion
```

whsieh commented on Source/WebKit/UIProcess/mac/WKTextIndicatorStyleManager.h:
```diff
@@ -0,0 +1,50 @@
+
+@interface WKTextIndicatorStyleManager : NSObject
+
+- (instancetype)initWithWebViewImpl:(WebKit::WebViewImpl*)view;
```
Nit - I think you can take a reference (`WebKit::WebViewImpl&`) since this `view` should never be `nullptr` when creating this object.

whsieh commented on Source/WebKit/UIProcess/mac/WKTextIndicatorStyleManager.h:
```diff
@@ -0,0 +1,50 @@
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
```
Extra newline here.

megangardner committed:
Add support for text indicator style for mac.
https://bugs.webkit.org/show_bug.cgi?id=272121
rdar://123134304

Reviewed by Wenson Hsieh.

Add support for text indicator style for mac.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _removeTextIndicatorStyleForID:]):
* Source/WebKit/UIProcess/mac/WKTextIndicatorStyleManager.h: Added.
* Source/WebKit/UIProcess/mac/WKTextIndicatorStyleManager.mm: Added.
* Source/WebKit/UIProcess/mac/WebViewImpl.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::addTextIndicatorStyleForID):
(WebKit::WebViewImpl::removeTextIndicatorStyleForID):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/277080@main
